diff --git a/.diff b/.diff
new file mode 100644
index 0000000..ec3908d
--- /dev/null
+++ b/.diff
@@ -0,0 +1,8 @@
+.env.development
+.env.staging
+.env.test
+README.md
+app/controllers/posts_controller.rb
+app/models/comment.rb
+config/application.rb
+lib/simplecov/git_diff_filter.rb
diff --git a/.env.development b/.env.development
index 68ba0a6..c3a2b92 100644
--- a/.env.development
+++ b/.env.development
@@ -1,2 +1,4 @@
-#TEST_COV=myTest1
+TEST_COV=myTest2
+DIFF_FILE_PATH=.diff
+#TARGET_BRANCH=branch1
 REDIS_URL=redis://127.0.0.1:6379/0
\ No newline at end of file
diff --git a/.env.staging b/.env.staging
new file mode 100644
index 0000000..e69de29
diff --git a/.env.test b/.env.test
deleted file mode 100644
index 1a15c7f..0000000
--- a/.env.test
+++ /dev/null
@@ -1,2 +0,0 @@
-TEST_COV=myTest2
-REDIS_URL=redis://127.0.0.1:6379/1
\ No newline at end of file
diff --git a/README.md b/README.md
index 7db80e4..f13bc51 100644
--- a/README.md
+++ b/README.md
@@ -20,5 +20,3 @@ Things you may want to cover:
 * Services (job queues, cache servers, search engines, etc.)
 
 * Deployment instructions
-
-* ...
diff --git a/app/controllers/posts_controller.rb b/app/controllers/posts_controller.rb
index 3f80c0b..345bfe2 100644
--- a/app/controllers/posts_controller.rb
+++ b/app/controllers/posts_controller.rb
@@ -71,4 +71,5 @@ class PostsController < ApplicationController
     def post_params
       params.require(:post).permit(:author, :title, :body)
     end
+
 end
diff --git a/app/models/comment.rb b/app/models/comment.rb
index 361e5cf..c8f5c5d 100644
--- a/app/models/comment.rb
+++ b/app/models/comment.rb
@@ -1,6 +1,6 @@
 class Comment < ApplicationRecord
 
-  validates_presence_of :commenter, :body
+  validates_presence_of :commenter
 
   belongs_to :post
 end
diff --git a/config/application.rb b/config/application.rb
index be163e0..55b5584 100644
--- a/config/application.rb
+++ b/config/application.rb
@@ -17,6 +17,11 @@ if ENV['TEST_COV']
     Redis.current.sadd(ENV.fetch('TESTS_KEY') { 'tests' }, ENV.fetch('TEST_COV'))
   end
 
+  if ENV['DIFF_FILE_PATH']
+    require_relative '../lib/simplecov/git_diff_filter'
+    SimpleCov.add_filter SimpleCov::GitDiffFilter.new(ENV['DIFF_FILE_PATH'])
+  end
+
   SimpleCov.start 'rails' do
     command_name "Manual Tests #{SecureRandom.uuid}"
     coverage_dir 'app/views/testings'
diff --git a/lib/simplecov/git_diff_filter.rb b/lib/simplecov/git_diff_filter.rb
new file mode 100644
index 0000000..f673821
--- /dev/null
+++ b/lib/simplecov/git_diff_filter.rb
@@ -0,0 +1,20 @@
+class SimpleCov::GitDiffFilter < SimpleCov::Filter
+
+  attr_reader :diff_files
+
+  def initialize(diff_file_path)
+    @diff_files = File.readlines(diff_file_path).map { |line| line.rstrip }
+  end
+
+  # def initialize(target_branch, base_branch = 'master')
+  #   diff = %x(git diff --name-only #{base_branch}..#{target_branch})
+  #   @diff_files = diff.split(/\n/)
+  # end
+
+  def matches?(source_file)
+    diff_files.none? do |arg|
+      source_file.filename =~ /#{arg}/
+    end
+  end
+
+end
\ No newline at end of file
.diff
.env.development
Gemfile
Gemfile.lock
config/application.rb
config/initializers/simplecov.rb
lib/simplecov-html/views/file_list.erb
lib/simplecov-html/views/source_file.erb
lib/simplecov/git_diff_filter.rb
